src/diem/jsonrpc/client.py

```diff
@@ -413,6 +413,28 @@ def get_account_state_with_proof(
         params = [address, version, ledger_version]
         return self.execute("get_account_state_with_proof", params, _parse_obj(lambda: rpc.AccountStateWithProof()))
 
+    def get_diem_id_domain_map(self) -> typing.Dict[str, str]:
+        domain_map = {}
+        event_index = 0
+        batch_size = 100
+        tc_account = self.must_get_account(utils.get_treasury_compliance_address())
+        event_stream_key = tc_account.role.diem_id_domain_events_key
+        while True:
+            events = self.get_events(event_stream_key, event_index, batch_size)
+            for event in events:
+                if event.data.removed:
+                    del domain_map[event.data.domain]
+                else:
+                    domain_map[event.data.domain] = event.data.address
+            if len(events) < batch_size:
+                break
+            event_index += batch_size
+        return domain_map
+
+    def get_vasp_address_with_domain(self, domain: str) -> typing.Optional[str]:
```

Comment:
I think we should remove this function, let caller to: `client.get_diem_id_domain_map().get(domain)`